src/shogun/classifier/LDA.cpp

```diff
@@ -154,8 +141,7 @@ bool CLDA::solver_classic()
 
 	auto solver = std::unique_ptr<LDASolver<ST>>(
 	    new LDASolver<ST>(
-	        dense_feat,
-	        new CMulticlassLabels(static_cast<CBinaryLabels*>(m_labels)),
+	        dense_feat, new CMulticlassLabels(binary_labels(m_labels)),
```

Comment:
pls lets not do this, but directly convert to multiclass

Reply:
You, all you need to do is to add a case in `multiclass_labels`, which deals with the case when `LT_BINARY` is provided. The conversion will be the same as in the constructor you mentioned (just using `std::transform`, and some checks). What is good about this is that **if** the user passed multiclass labels in the first place, there would no conversion. If the user passes binary labels, everything is as before (nonsense conversion, but we can leave that for now as we are working on a different topic here)

I don't understand what you mean re the contiguous labels.

Reply:
yes we can take care of binary and multiclass easily however if we pass dense labels we will need the std::transform conversion (like in binary) too. That is do the same thing we do with binary labels (converting ```[-1, 1]``` to ```[0, 1]```)except now the labels are dense. This is not already present in multiclass_labels but instead we have a require statement that checks the dense labels to be of type ```[0, 1...]``` (this is why we needed to convert first to binary and then  to multiclass earlier) what we could now is replace the require statement with a loop that does exactly that with discrete valued dense labels.